pkg/controller/groups/group.go

```diff
@@ -139,17 +144,50 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
 }
 
-func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
 	cr, ok := mg.(*v1alpha1.Group)
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errNotGroup)
 	}
-	_, _, err := e.client.UpdateGroup(
+	grp, _, err := e.client.UpdateGroup(
 		meta.GetExternalName(cr),
 		groups.GenerateEditGroupOptions(cr.Name, &cr.Spec.ForProvider),
 		gitlab.WithContext(ctx),
 	)
-	return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateFailed)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateFailed)
+	}
+
+	if len(cr.Spec.ForProvider.SharedWithGroups) > 0 {
+		for _, sh := range cr.Spec.ForProvider.SharedWithGroups {
+			if notShared(sh.GroupID, grp) {
+				opt := gitlab.ShareGroupWithGroupOptions{
+					GroupID:     sh.GroupID,
+					GroupAccess: (*gitlab.AccessLevelValue)(&sh.GroupAccessLevel),
+				}
+				if sh.ExpiresAt != nil {
+					opt.ExpiresAt = (*gitlab.ISOTime)(&sh.ExpiresAt.Time)
+				}
+				_, _, err = e.client.ShareGroupWithGroup(grp.ID, &opt)
+				if err != nil {
+					return managed.ExternalUpdate{}, errors.Wrap(err, errShareFailed)
```

Comment:
Maybe use `errors.Wrapf` to return a formatted error message that includes `sh.GroupID` that is currently used. Ref [Golang String Formatting](https://pkg.go.dev/fmt)